lib/kafka_ex/protocol/sync_group.ex

```diff
@@ -0,0 +1,66 @@
+defmodule KafkaEx.Protocol.SyncGroup do
+  import KafkaEx.Protocol.Common
+
+  @member_assignment_version 0
+
+  defmodule Response do
+    defstruct error_code: nil, assignments: []
+    @type t :: %Response{error_code: atom | integer, assignments: [{binary, [integer]}]}
```

Comment:
Well, given that error codes are ran through `KafkaEx.Protocol.error` which has a `atom | integer` signature, there's not much to do here (or anywhere else, I'd say. I wonder why dialyzer let it through).

+1 on the type aliases. Here and everwhere else btw ;-)